pkg/provisioner/provision.go

```diff
@@ -24,8 +24,10 @@ func (p *ZFSProvisioner) Provision(ctx context.Context, options controller.Provi
 	datasetPath := fmt.Sprintf("%s/%s", parameters.ParentDataset, options.PVName)
 	properties := make(map[string]string)
 
-	if parameters.NFS != nil {
-		properties["sharenfs"] = parameters.NFS.ShareProperties
+	useHostPath := true
+	if parameters.Type == "nfs" || parameters.Type == "auto" && (options.SelectedNode == nil || parameters.HostPathNodeName != options.SelectedNode.Name) {
+		useHostPath = false
+		properties["sharenfs"] = parameters.NFSShareProperties
```

Comment:
It's been a while since I last programmed in Go, is this not possible?
```suggestion
		properties[SharePropertiesParameter] = parameters.NFSShareProperties
```